### PR TITLE
Bug 1868940 - Cannot install an add-on a from local file on Android <10.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1868940

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -711,7 +711,9 @@ class SettingsFragment : PreferenceFragmentCompat() {
     @VisibleForTesting
     internal fun setupInstallAddonFromFilePreference(settings: Settings) {
         with(requirePreference<Preference>(R.string.pref_key_install_local_addon)) {
-            isVisible = settings.showSecretDebugMenuThisSession
+            // Below Android 10, the OS doesn't recognize the web extension mime type.
```
Comment (willdurand) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt:714]:
```suggestion
            // Below Android 10, the OS doesn't seem to recognize the "application/x-xpinstall" mime type (for XPI files).

File: fenix/app/src/test/java/org/mozilla/fenix/settings/SettingsFragmentTest.kt [hunk 2]
```diff
@@ -98,6 +99,7 @@ class SettingsFragmentTest {
     }
 
     @Test
+    @org.robolectric.annotation.Config(sdk = [Build.VERSION_CODES.Q])
```
Comment (willdurand) [inline fenix/app/src/test/java/org/mozilla/fenix/settings/SettingsFragmentTest.kt:102]: Could we have a test to verify that the setting is hidden when we have a version code for Android 9 or something?

**Review by willdurand — APPROVED** [review]
r+wc, thanks - please add a short test :)

File: fenix/app/src/test/java/org/mozilla/fenix/settings/SettingsFragmentTest.kt (hunk 2 above)
Comment (Amejia481) [reply fenix/app/src/test/java/org/mozilla/fenix/settings/SettingsFragmentTest.kt:102]: Added :)

**Pull request merged**